### DISCUSSION TITLE
Local model seems non-local

### DISCUSSION THREAD

**Discussion by choi328328** [NVIDIA/NVFlare#694] [Q&A] [answered]
Hi. My name is Byungjin Choi, and I am a Medical informatician in Korea.
Thank you for constructing such an excellent framework.
However, there are some things I can't understand in the hello-pt-tb example.

I think the nvflare works like below: 

Scatter_and_Gather flow start:
    for each round:
            1)	Get the global model(weight) from the server
            2)	Local train
            3)	Save local model
            4)	Send weight to the server
Scatter_and_Gather Done

CrosssiteEval flow start : 
	In each client’s validation data:
		Valiate Global model 
		Validate each client’s model
	Report results 

If I understand correctly, the 'local model' is not trained only with local data.
Instead, it is a fine-tuned (or local adaptation) model of the global model(n-1 round).

Wouldn't it be more appropriate to compare the FL global model with the model trained only on local data?

**Comment by holgerroth** [comment, marked as answer]
You are correct in your observation. The local model is a "personalized" or "fine-tuned" version of the previous global model. Depending on the implementation of the Learner, we could also do some model selection based on the local validation scores. The `CIFAR10Learner` is doing that for example [here](https://github.com/NVIDIA/NVFlare/blob/243b8fc9e083d837935ef607437f18dd72cae858/examples/cifar10/pt/learners/cifar10_learner.py#L244).

By setting the `num_rounds=1` in [config_fed_server.json](https://github.com/NVIDIA/NVFlare/blob/dev/examples/hello-pt-tb/config/config_fed_server.json), you can basically simulate each client only training on their local data (that's the current default in the hello-pt-tb example. Also increase `epochs` in [config_fed_client.json](https://github.com/NVIDIA/NVFlare/blob/dev/examples/hello-pt-tb/config/config_fed_client.json) to do a complete local training.

**Reply by choi328328** [reply]
Thank you for your kind reply! My question is resolved.
